sdks/python/apache_beam/typehints/opcodes.py

```diff
@@ -440,18 +450,30 @@ def make_function(state, arg):
   """
   # TODO(luke-zhu): Handle default argument types
   globals = state.f.__globals__  # Inherits globals from the current frame
-  func_name = state.stack[-1].value
-  func_code = state.stack[-2].value
-  pop_count = 2
+  tos = state.stack[-1].value
+  if isinstance(tos, types.CodeType):
+    func_name = None
```

Comment:
This is related to the lack of a fully qualified name being pushed onto the stack for lambdas in 3.11 on. For pre-3.11 versions TOS in calls to `make_function()` will always be a string containing the fully qualified name of the function being made, but post-3.11 this could just be the code for the lambda. We have to check for this case and adjust the indices we access on the stack accordingly to avoid index-out-of-range issues